### ISSUE TITLE
Support SQLDataType.XML in DDL

### ISSUE THREAD

**Issue by AnghelLeonard** [jOOQ/jOOQ#11331]
### Expected behavior
Using `SqlDataType.XML` should render `XMLTYPE` for Oracle.

### Actual behavior
Using `SqlDataType.XML` renders `XML` for Oracle and causes  ORA-00902: invalid datatype.

### Steps to reproduce the problem
Simply use `json_table` and:

```Java
...
.column("name", VARCHAR)   // ok
.column("code", VARCHAR)   // ok
.column("capacity", XML)   // not ok
...
```

### Versions

- jOOQ:3.14.4
- Java:14
- Database (include vendor): Oracle 18c
- OS: Windows 10
- JDBC Driver (include name if inofficial driver): ojdbc8

**Comment by lukaseder** [comment]
Thanks a lot for your report. I don't think we have integration tests for that yet, which is an oversight. Meaning there might be other dialects where it doesn't work yet. It probably works by accident in PostgreSQL, which has an `XML` type.

**Comment by lukaseder** [comment]
Looking into this now. Related issue: https://github.com/jOOQ/jOOQ/issues/13252

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0. Thanks again for your report!

**Comment by lukaseder** [comment]
This works correctly in Db2 and PostgreSQL, so it's an Oracle-only issue.
